numpy/core/src/multiarray/methods.c

```diff
@@ -1626,6 +1625,79 @@ array_reduce(PyArrayObject *self, PyObject *NPY_UNUSED(args))
     return ret;
 }
 
+static PyObject *
+array_reduce_ex(PyArrayObject *self, PyObject *args)
+{
+    int protocol;
+    PyObject *ret = NULL, *numeric_mod = NULL, *from_buffer_func = NULL;
+    PyObject *buffer_tuple = NULL, *pickle_module=NULL, *pickle_class = NULL;
+    PyObject *class_args = NULL, *class_args_tuple = NULL, *unused = NULL;
+    PyArrayObject *buffer = NULL;
+    PyArray_Descr *descr = NULL;
+
+    if (PyArg_ParseTuple(args,"i", &protocol)){
+        if (protocol == 5){
+            ret = PyTuple_New(2);
+
+            if (ret == NULL) {
+                return NULL;
+            }
+
+            /* if the python version is below 3.8, the pickle module does not provide
+             * built-in support for protocol 5. We try importing the pickle5
+             * backport instead */
+            if (PY_VERSION_HEX < 0x03080000){
```

Comment:
Rather than a runtime condition, it should be possible to use a build time preprocessor directive here.

Reply:
Yes, altough it was not necessary for the code to compile - definitely can do it in further iterations, but did not want to generate any misunderstandings :)